### PR TITLE
Add promise rejection tracker support

### PR DESCRIPTION
**Pull request by rerobika**
JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

**Comment by rerobika** [comment]
Related feature request: #2931 .

**Comment by dbatyai** [comment]
I feel like it would be better to handle this as a port function instead of a callback that can be set via the API.

File: tests/unit-core/test-promise.c [hunk 1]
```diff
@@ -158,6 +174,34 @@ main (void)
...
+  TEST_ASSERT (!jerry_value_is_error (my_promise4));
+  TEST_ASSERT (tracter_counter == 2);
+  TEST_ASSERT (jerry_get_boolean_value (jerry_binary_operation (JERRY_BIN_OP_STRICT_EQUAL,
+                                                                last_rejected_promise,
+                                                                my_promise4)));
+  TEST_ASSERT (jerry_value_is_object (last_rejected_reason));
+  TEST_ASSERT (jerry_get_error_type (last_rejected_reason) == JERRY_ERROR_TYPE);
+  jerry_release_value (last_rejected_promise);
+  jerry_release_value (last_rejected_reason);
+
```
Comment (galpeter) [inline tests/unit-core/test-promise.c:204]: There should be also a test where the "global" reject handler is not invoked as the reject was handled already.

File: tests/unit-core/test-promise.c [hunk 2]
```diff
@@ -30,8 +30,11 @@ static const jerry_char_t test_source[] = TEST_STRING_LITERAL (
 );
 
 static int count_in_assert = 0;
+static int tracter_counter = 0;
```
Comment (galpeter) [inline tests/unit-core/test-promise.c:33]: What does "tracter" mean?

File: docs/02.API-REFERENCE.md [hunk 3]
```diff
@@ -7692,6 +7716,67 @@ main (void)
 - [jerry_run](#jerry_run)
 - [jerry_vm_exec_stop_callback_t](#jerry_vm_exec_stop_callback_t)
 
+## jerry_set_promise_rejection_callback
+
+**Summary**
+
+Register a promise rejection tracker callback for the engine.
```
Comment (galpeter) [inline docs/02.API-REFERENCE.md:7723]: We should provide more details here imho and also mention the `HostPromiseRejectionTracker` from the standard here.

**Comment by rerobika** [comment]
The PR should be reworked as a port function, so closing the PR for now.

**Pull request closed without merging**